### PR TITLE
Do not explicitly initialize memory objects to zero.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by leonardoalt — APPROVED** [review]

**Review by leonardoalt — APPROVED** [review]

**Comment by lvella** [comment]
How does this PR ensures that a read on uninitiated memory will return 0?

**Comment by leonardoalt** [comment]
That's just how the current memory state machine works

File: riscv/src/compiler.rs [hunk 1]
```diff
@@ -176,8 +183,9 @@ fn store_data_objects<'a>(
                 }
                 pos += item.size() as u32;
             }
-            let first_line = object_code.first_mut().unwrap();
```
Comment (leonardoalt) [inline riscv/src/compiler.rs:179]: Why did this not panic before?

**Comment by lvella** [comment]
when merge?

File: riscv/src/compiler.rs (hunk 1 above)
Comment (chriseth) [reply riscv/src/compiler.rs:179]: `object_code` was not empty before, now it can because `DataValue::Zero` does not add anything to `object_code`.

**Pull request merged**